src/Views/AppInfoView.vala

```diff
@@ -1049,7 +1066,13 @@ public class AppCenter.Views.AppInfoView : AppCenter.AbstractAppContainer {
 
                         release_carousel.add (release_row);
 
+#if HAS_APPSTREAM_1_0
+                        var release_version = new AppStream.Relation ();
+                        release_version.set_version (release.get_version ());
+                        if (package.installed && release_version.version_compare (package.get_version ())) {
```

Comment:
Haven't tested if this compiles or works, but this feels closer to the original:

```suggestion
                        if (package.installed && AppStream.vercmp_simple (release.get_version (), package.get_version ()) <= 0) {
```